tls/handshake.py

```diff
@@ -59,383 +55,443 @@ def x509_check_issuer(cert, issuer):
-        if addr:
-            self.addr = addr
-        else:
-            self.addr = tf_cfg.cfg.get("Tempesta", "ip")
-        self.port = port
-        self.io_to = io_to  # seconds, maybe not so small fraction of a second.
-        self.chunk = chunk
-        self.sleep_time = sleep_time if sleep_time >= 0.001 else 0.001
-        # We should be able to send at least 10KB with 1ms chunk delay for RTO.
-        if self.chunk and self.chunk < 10000:
-            io_to = 10000 / self.chunk * 0.001
-            if self.io_to < io_to:
-                self.io_to = io_to
-        self.verbose = verbose
-        # Service members.
-        self.sock = None
-        # Additional handshake options.
-        self.sni = ["tempesta-tech.com"]  # vhost string names.
-        self.exts = []  # Extra extensions
-        self.sign_algs = []
-        self.elliptic_curves = []
-        self.ciphers = []
-        self.compressions = []
-        self.renegotiation_info = []
-        self.inject = None
-        # HTTP server response (headers and body), if any.
-        self.http_resp = None
-        # Host request header value, taken from SNI by default.
-        self.host = None
-        # Server certificate.
-        self.cert = None
-        # Random session ticket by default.
-        self.set_ticket_data("ticket_data")
-        # Session id, must be filled for resume
-        self.session_id = ""
-
-    def set_ticket_data(self, data):
-        """Set session ticket data. Following values are possible:
-        - 'None' - session ticket extension will be disabled;
-        - '' (empty string) - empty session ticket extension will be added;
-        - '<str>' - arbitrary string will be inserted as session ticket.
-        'data' can be represented either as string or as
-        scapy_ssl_tls.TLSSessionTicket.
-        """
-        if data is None:
-            self.ticket_data = None
-            return
-        if type(data) is tls.TLSSessionTicket:
-            self.ticket_data = data.ticket
-        else:
-            self.ticket_data = data
-
-    @contextmanager
-    def socket_ctx(self):
-        try:
-            yield
-        finally:
-            self.sock.close()
-
-    def conn_estab(self):
-        assert not self.sock, "Connection has already been established"
-        self.sock = tls.TLSSocket(socket.socket(), client=True)
-        # Set large enough send and receive timeouts which will be used by
-        # default.
-        self.sock.settimeout(self.io_to)
-        self.sock.setsockopt(
-            socket.SOL_SOCKET, socket.SO_RCVTIMEO, struct.pack("ll", int(self.io_to * 1000), 0)
+    
+    def __init__(self, *args, **kwargs):
+        self.cached_secrets = kwargs.pop("cached_secrets", None)
+        self.chunk = kwargs.pop("chunk", None)
+        self.send_data = []
+        self.server_data = []
+        self.hs_state = False
+        self.hs_final = False
```

Comment:
How is this flag used? I see that it is only set and never checked.

Reply:
Fixed, its unused variable